src/openrct2/drawing/NewDrawing.cpp

```diff
@@ -209,6 +211,28 @@ void gfx_draw_line(rct_drawpixelinfo* dpi, int32_t x1, int32_t y1, int32_t x2, i
     }
 }
 
+void gfx_draw_dashed_line(
+    rct_drawpixelinfo* dpi, const int32_t x1, const int32_t y1, const int32_t x2, const int32_t y2,
+    const int32_t dashedLineSegmentLength, const int32_t colour)
+{
+    const auto drawingEngine = dpi->DrawingEngine;
+    if (drawingEngine != nullptr)
+    {
+        const auto dashedLineLength = hypot(x2 - x1, y2 - y1);
+        const auto lineSegmentCount = dashedLineLength / dashedLineSegmentLength / 2;
+        const auto angle = atan2(y2 - y1, x2 - x1);
+        const auto dx = dashedLineSegmentLength * cos(angle);
+        const auto dy = dashedLineSegmentLength * sin(angle);
```

Comment:
So far we have avoided floating point numbers everywhere in the game apart from the OpenGL coding. Is it possible for you to make this integer based?

Reply:
Don't these get coerced to integers later, anyways?

Reply:
Doesn't matter, the intermediate step with floats is enough to introduce inconsistencies across clients.

If possible, please avoid them.

Reply:
> introduce inconsistencies across clients.

This is drawing code so that isn't a problem. But it is just that we have done so well to avoid them, we would like to keep that if possible for the software renderer.

Using triganometry for a dashed line seems overkill anyway. You should just be able to just work out the width and height ratios for each dash. Multiply them up to large integers and then down again to avoid losing precision.

Reply:
I tested diagonals with your idea of multiplying big and going back down to avoid losing precision. It seems to work decently, so I will commit that.